assets/js/components/Dashboard/PaMessaging/NewPaMessagePage/NewPaMessagePage.tsx

```diff
@@ -22,9 +23,27 @@ const NewPaMessagePage = () => {
   const [interval, setInterval] = useState("4");
   const [visualText, setVisualText] = useState("");
   const [phoneticText, setPhoneticText] = useState("");
+  const [audioPlaying, setAudioPlaying] = useState(false);
 
   const navigate = useNavigate();
 
+  const previewAudio = () => {
+    setAudioPlaying(true);
+    fetchAudioPreview(phoneticText.length ? phoneticText : visualText).then(
+      async (audioData) => {
+        const blob = await audioData.blob();
+        const url = URL.createObjectURL(blob);
+        const audio = new Audio(url);
```

Comment:
It feels like we should be able to streamline this by letting the browser itself fetch the audio (constructing the `Audio` by passing in a URL like `/preview_audio?text=Here%20is%20my%20message`)... though I don't think there's a way to have this use a non-GET request, which means the length of the text we could preview would be limited by the length of a URL. Still, a URL can be around 2000 characters (8000 if you ignore Edge), which feels like it should be more than enough for a PA message. What do you think of this possibility?

Reply:
I like this idea. The length requirement doesn't bother me because messages that exceed that length are probably too long to play anyway. I'd hate to add that without the user having knowledge of an actual limit, though. Maybe I should check with Mary/Jim and see if we should just make the decision to add a limit to the textboxes?

Reply:
If we had a real-time character count a la Twitter it would jump around a lot, since non-URL-safe characters (most notably spaces) would take up more than one character. We could at least prevent entering text that would take the input over the limit, and given how unlikely this is, maybe that's enough. I'd ask if having the limit is okay, of course, but I don't see why it wouldn't be.